### DISCUSSION TITLE
Extending theme types

### DISCUSSION THREAD

**Discussion by leogilmour** [heroui-inc/heroui#877] [Help] [answered]
*Not sure if this should be in "Help" or "Feature requests"*

<img width="734" alt="image" src="https://user-images.githubusercontent.com/5797658/201672339-6ea7d6e5-a520-458f-b94b-4069241f5a79.png">

- I have created a new color in our theme. When I need to reference this color (i.e. I can't use it in the css object with the dollar sign `$primaryDarkIcon`) I get a typescript error because this does not exist on the default declared stitches color types.

- Either this is a "help" question, and I'd love some advice about how best to go around adding this as a type to `theme.colors`
- Or this is a feature request: is there a way types can be dynamic and include our new color key values ?

[Customize Theme Docs](https://nextui.org/docs/theme/customize-theme)

**Comment by jrgarciadev** [comment, marked as answer]
Hey @leogilmour unfortunately custom colours don't have typescript suggestions, we've been working on improving the theme extension and typescript support, you can use your custom colour though https://codesandbox.io/s/custom-color-autocomplete-p7y933?file=/App.js

**Reply by leogilmour** [reply]
Hi @jrgarciadev Thanks for getting back to me :) 
Ok cool. I guess I should look at creating my own type subset and extending the themes with my additions.

Yeah the `$` works great. And my custom colour **does** work, it's just the typescript error to ignore. Currently doing this (for another library's components):

![image](https://user-images.githubusercontent.com/5797658/202147264-bf1f57d5-060f-4b6b-a4e2-547782925990.png)

Which isn't ideal if reusing multiple times, so will look into extending the theme types.

**Reply by corydeppen** [reply]
It's been a little while since this was marked as answered but it seems the issue/limitation still exists and I wasn't able to find an open issue or PR. @jrgarciadev Is there work in progress that you can point us to if we want to follow along?
